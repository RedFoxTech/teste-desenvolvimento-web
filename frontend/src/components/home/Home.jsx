import React from 'react'
import Main from '../template/Main'

export default props =>
    <Main icon="home" title="Home"
        subtitle="Pokemón Project of Red Fox Tech">
        <div className="h1 display-4">Welcome!
        </div>
        <hr />
        <p className="mb-0 lead">System to take control of the Pokemóns!</p>
    </Main>