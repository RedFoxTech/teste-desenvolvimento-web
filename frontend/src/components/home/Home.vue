
<template>

  <div>
    <h1 class="title">{{ Nome }}</h1>

    <input type="search" class="input is-rounded" v-on:input="filtro = $event.target.value" placeholder="Procure o Pokemon....">

    <ul class="list-pokemons">
      <li class="list-pokemon" v-for="pokemon of searchPokemon">

        <meu-painel :Nome="pokemon.Nome" :Pokedex="pokemon.PokedexNumber">
          <img class="img" src="https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcQwtlnmqL44rQVFJc1vBMT_p7tjT6opPTHhqw&usqp=CAU" :alt="pokemon.Nome">
      
        </meu-painel>

      </li>
    </ul>
  </div>

</template>

<script>
import axios from 'axios';
import Painel from '../shared/painel/Painel.vue';

export default {

   components: {
     'meu-painel' : Painel
   },

   data() {
     return{

       Nome:'Pokedex',
       pokemons: [],
       filtro: ''
    
     }
   },

   computed: {
     searchPokemon() {

       if(this.filtro) {
         let exp = new RegExp(this.filtro.trim(), 'i');
         return this.pokemons.filter(pokemon => exp.test(pokemon.Nome));
       }else{
         return this.pokemons;
       }
     }
   },
   
   created (){
    axios.get("http://localhost:3000").then(res => {
      console.log(res.data);
      this.pokemons = res.data;
      console.log(this.pokemons);
      
    })
  },
}
</script>

<style>

    .title {
      text-align: center;
    }

    .list-pokemons {
      list-style: none;
    }

    .list-pokemons .list-pokemon {
      display: inline-block;

    }

    .list-pokemons .list-pokemon .img {
      width: 100%;
    }
</style>