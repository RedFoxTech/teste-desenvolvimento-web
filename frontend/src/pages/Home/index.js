import React, { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { Navbar, Nav, Form, FormControl, FormGroup, Container, Card, Row, Col, Button } from 'react-bootstrap';
import { FaArrowLeft, FaSearch } from 'react-icons/fa';

import { pad } from '../../utils/pokedex_number';

import pokeLogo from '../../assets/logo-dex.png';

import './styles.css';
import api from '../../services/api';

const Home = () => {
    const [pokemons, setPokemons] = useState([]);
    const [counter, setCounter] = useState(0);
    const lastPage = Math.ceil(counter / 20);

    const [search, setSearch] = useState('');

    const [page, setPage] = useState(1);

    async function getPokemons() {
        await api.get(`/pokemons?page=${page}`)
            .then(res => {
                setPokemons(res.data[1]);
                setCounter(res.data[0].total);
            }).catch(err => {
                alert(err);
            });
    }

    useEffect(() => {
        getPokemons();
    }, [page]);

    async function Procurar() {
        await api.post('/search', { SearchWord: search })
            .then(res => {
                if (res.data) {
                    setPokemons(res.data);
                }
            }).catch(err => {
                alert(err);
            })
    }

    useEffect(() => {
        if (search !== '') {
            Procurar();
        } else {
            getPokemons();
        }
    }, [search]);

    return (

        <div id="home">
            <header className="header-menu">
                <Navbar expand="lg" variant="dark">

                    <Link className="navbar-brand" to="/">
                        <img
                            src={pokeLogo}
                            alt="Pokemon-Logo"
                            width={200}
                        />
                    </Link>
                    <Navbar.Toggle aria-controls="responsive-navbar-nav" />

                    <Nav className="w-100 justify-content-end">
                        <Link to="/" className="nav-link active">Página inicial</Link>
                        <Link to="/login" className="nav-link">Login / Registro</Link>
                    </Nav>
                </Navbar>
            </header>

            <Form onSubmit={() => { }} inline className="search-box d-flex justify-content-center">
                <FormGroup>
                    <FormControl type="text" placeholder="Buscar ID ou Nome" value={search} onChange={e => setSearch(e.target.value)} />
                    <FaSearch size={22} className="text-warning search-icon" />
                </FormGroup>
            </Form>
            
            <Container className="text-light mt-4 mb-4">
                <Row >
                    {pokemons.map(pokemon => (
                        <Col
                            xs={12}
                            sm={6}
                            md={4}
                            lg={3}
                            xl={3}
                            className="text-light"
                            key={pokemon.id}
                            className="mb-3"
                        >
                            <Card className="text-light bg-dark">
                                <Card.Body>
                                    <Card.Title className="text-center">{pokemon.Name}</Card.Title>
                                    <img
                                        src={"https://assets.pokemon.com/assets/cms2/img/pokedex/detail/" +
                                            pad(pokemon.Pokedex_Number, 3) + ".png"}
                                        alt="pokemon_img"

                                    />
                                </Card.Body>
                            </Card>
                        </Col>
                    ))}
                </Row>
            </Container>
            {counter > 20 && search === '' ?
                <Container>
                    <Row className='justify-content-center'>
                        {page !== 1 ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(page - 1)}>
                                <FaArrowLeft />
                            </Button>
                            :
                            null
                        }

                        {page - 2 > 0 ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(page - 2)}>
                                {page - 2}
                            </Button>
                            :
                            null
                        }

                        {page - 1 > 0 ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(page - 1)}>
                                {page - 1}
                            </Button>
                            :
                            null
                        }

                        <Button variant='secondary' className='ml-3'>
                            {page}
                        </Button>

                        {page + 1 < lastPage ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(page + 1)}>
                                {page + 1}
                            </Button>
                            :
                            null
                        }

                        {page + 2 < lastPage ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(page + 2)}>
                                {page + 2}
                            </Button>
                            :
                            null
                        }

                        {page !== lastPage ?
                            <Button variant="warning" className="ml-3" onClick={() => setPage(lastPage)}>
                                Ultima página
                            </Button>
                            :
                            null
                        }


                    </Row>
                </Container>
                :
                null
            }
        </div>
    )
}
export default Home;