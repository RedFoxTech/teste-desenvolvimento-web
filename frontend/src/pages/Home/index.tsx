import React, { useEffect, useState, MouseEvent} from 'react';
import { Navbar, Button, Form, Figure } from 'react-bootstrap';
import { Link, useHistory } from 'react-router-dom';
import logo from '../../assets/pokeball.svg';
import { FiArrowLeft, FiArrowRight } from 'react-icons/fi';
import './styles.css';
import api from '../../services/api';


const Home = () => {
    const history = useHistory();
    interface Pokemon {
        id: number,
        name: string,
        pokedex_number: number,
        image: string,
        generation: number,
        evolution_stage: number,
        family_id: number,
        atk: number,
        def: number,
        sta: number,
        max_cp_at_40: number,
        max_cp_at_39: number,
        type1: number,
        type2: number,
        weather1: number,
        weather2: number,
        image_type1: string,
        name_type1: string,
        image_type2: string,
        name_type2: string,
        image_url: string,
        image_url_type1: string,
        image_url_type2: string,
        image_url_weather1: string,
        image_url_weather2: string
    }


    const [pokemons, setPokemons] = useState<Pokemon[]>([]);
    const [pages, setPage] = useState<number>(1);
    const [filters, setFilter] = useState<string>('');
    const rows = 9;
    useEffect(() => {
        api.get(`pokemons?filter=${filters}&pages=${pages}&rows=${rows}`).then(response => {
            console.log(response.data);
            setPokemons(response.data);
        })
    }, []);

    let aux = 0;

    async function nextPage() {
        const page = pages + 1;
        await api.get(`pokemons?filter=${filters}&pages=${page}&rows=${rows}`).then(response => {
            if (response.data[0] === undefined)
                return aux = 1;
            setPokemons(response.data);
            aux = 0;
        });
        if (aux === 1)
            return;
        setPage(pages + 1);
    }

    async function previousPage() {
        if (pages === 1)
            return;
        const page = pages - 1;
        setPage(pages - 1);
        console.log(pages);
        api.get(`pokemons?filter=${filters}&pages=${page}&rows=${rows}`).then(response => {
            console.log(response.data);
            setPokemons(response.data);
        });
    }

    async function handleDeletePokemon(event: MouseEvent<HTMLButtonElement>) {
        const id = Number(event.currentTarget.value);
        console.log(id);
        try {
            await api.delete(`pokemons/${id}`);
            setPokemons(pokemons.filter(pokemon => pokemon.id !== id));
        } catch (err) {
            alert('Error in delete!');
        }

    }

    async function handleUpdatePokemon(event: MouseEvent<HTMLButtonElement>) {
        const id = event.currentTarget.value;
        localStorage.clear();
        localStorage.setItem('pokemon_id', id);
        history.push('/update-pokemon');

    }

    async function handleSearchPokemon(event: React.ChangeEvent<HTMLInputElement>) {
        const filter = String(event.currentTarget.value);
        let page0 = 1;
        setPage(1);

        setFilter(filter);
        api.get(`pokemons?filter=${filter}&pages=${page0}&rows=${rows}`).then(response => {
            console.log(response.data);
            setPokemons(response.data);
        });
        console.log(filters);
    }

    return (
        <div id="pokemon-profile">
            <Navbar bg="dark">
                <Navbar.Brand href="/">
                    <img
                        src={logo}
                        width="60"
                        height="60"
                        className="d-inline-block align-top"
                        alt="React Bootstrap logo"
                    />
                </Navbar.Brand>
                <Navbar.Toggle />
                <Navbar.Collapse className="justify-content-end">
                    <Link to="/create-pokemon">
                        <Button >Cadastrar Pokemon </Button>
                    </Link>
                </Navbar.Collapse>
            </Navbar>
            <Form.Control className="txtSearch" type="text" placeholder="Digite o Pokemon para pesquisa!" onChange={handleSearchPokemon} />
            <div className="pokemon-container">
                <ul>
                    {
                        pokemons.map(pokemon => (


                            <li key={pokemon.id}>
                                <ul>
                                    <li>
                                        <strong>Nome</strong>
                                        <p>{pokemon.name}</p>
                                        <strong>Generation</strong>
                                        <p>{pokemon.generation}</p>
                                        <strong>Family ID</strong>
                                        <p>{pokemon.family_id}</p>
                                        <strong>Evolution Stage</strong>
                                        <p>{pokemon.evolution_stage}</p>
                                        <strong>Cp 100% @ 40</strong>
                                        <p>{pokemon.max_cp_at_40}</p>
                                        <strong>Cp 100% @ 39</strong>
                                        <p>{pokemon.max_cp_at_39}</p>
                                    </li>
                                    <li style={{ width: 250, height: 300 }}>
                                        <Figure.Image
                                            width={250}
                                            height={300}
                                            src={pokemon.image_url}
                                            style={{ background: "#cefcb1", borderRadius: 5 }}
                                        />
                                        <strong>Pokedex Number: {pokemon.pokedex_number}</strong>
                                        <strong>Types:</strong>
                                        <ul>
                                            <li style={{ height: 50, marginTop: -25, background: "transparent", marginBottom: 25 }}>
                                                <Figure.Image
                                                    width={50}
                                                    height={50}
                                                    src={pokemon.image_url_type1}
                                                    style={{ background: "#dedede", borderRadius: 5 }}
                                                />
                                            </li>
                                            <li style={{ height: 50, marginTop: -25, background: "transparent" }}>
                                                <Figure.Image
                                                    width={50}
                                                    height={50}
                                                    src={pokemon.image_url_type2}
                                                    style={{ background: "#dedede", borderRadius: 5 }}
                                                />
                                            </li>
                                        </ul>
                                        <strong>Weather:</strong>
                                        <ul>
                                            <li style={{ height: 50, marginTop: -25, background: "transparent", marginBottom: 25 }}>
                                                <Figure.Image
                                                    width={50}
                                                    height={50}
                                                    src={pokemon.image_url_weather1}
                                                    style={{ background: "#dedede", borderRadius: 5 }}
                                                />
                                            </li>
                                            <li style={{ height: 50, marginTop: -25, background: "transparent", marginBottom: 25 }}>
                                                <Figure.Image
                                                    width={50}
                                                    height={50}
                                                    src={pokemon.image_url_weather2}
                                                    style={{ background: "#dedede", borderRadius: 5 }}
                                                />
                                            </li>
                                        </ul>
                                    </li>
                                </ul>


                                <ul className="Stats">
                                    <li style={{ background: "#85d17b" }}>
                                        <strong>ATK</strong>
                                        <p className="ATK">{pokemon.atk}</p>
                                    </li>
                                    <li style={{ background: "#7bc7d1" }}>
                                        <strong>DEF</strong>
                                        <p>{pokemon.def}</p>
                                    </li>
                                    <li style={{ background: "#d17b7b" }}>
                                        <strong>STA</strong>
                                        <p>{pokemon.sta}</p>
                                    </li>
                                    <li style={{ background: "#e6ed5c" }}>
                                        <strong>STAT TOTAL</strong>
                                        <p>{pokemon.atk + pokemon.def + pokemon.sta}</p>
                                    </li>
                                </ul>
                                <ul className="bottom-buttons">
                                    <li className="bottom-del">
                                        <button type="submit" className="deletebutton" onClick={handleDeletePokemon} value={pokemon.id}>Delete</button>
                                    </li>
                                </ul>
                            </li>
                        ))
                    }
                </ul>
            </div>
            <div className="navigationbar">
                <ul className="navigation">
                    <li>
                        <button type="submit" className="previousButton" onClick={previousPage}>
                            <FiArrowLeft />
                        </button>
                    </li>
                    <li>
                        <button type="submit" className="nextButton" onClick={nextPage} >
                            <FiArrowRight />
                        </button>
                    </li>
                </ul>
            </div>

        </div>
    );
}

export default Home;