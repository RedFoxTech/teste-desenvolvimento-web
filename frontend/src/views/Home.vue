<template>
  <pokemon />
</template>

<script>
  import Pokemon from '../components/Pokemon'

  export default {
    name: 'Home',

    components: {
      Pokemon,
    },
  }
</script>
