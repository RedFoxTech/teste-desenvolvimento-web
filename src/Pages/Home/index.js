import React, { Component } from 'react';
import Main from 'Components/main';
import Pagination from 'Components/Pagination';

// import api from 'services/api';
import {
	Col,
	Row,
	Container,
	Image,
	InputGroup,
	FormControl,
	Accordion,
	Button,
	Card,
} from 'react-bootstrap';
import Flip from 'react-reveal/Flip';

import { Type, Pokemons, Types, Search, Filter, BtnFilter } from './style';
import { Link } from 'react-router-dom';
import { pad } from 'assets/scripts';

import api from 'services/api';

export default class Home extends Component {
	constructor(props) {
		super(props);
		this.state = {
			data: [],
			rows: 0,
			searchWord: '',
			type: '',
		};
		this.search = this.search.bind(this);
		this.typeFilter = this.typeFilter.bind(this);
		this.clear = this.clear.bind(this);
		this.clearWord = this.clearWord.bind(this);
		this.clearType = this.clearType.bind(this);
	}

	componentDidMount() {
		api.get('/').then(response => {
			this.setState({ data: response.data.rows, rows : response.data.count });
		});
	}

	search(input) {
		const { id, value } = input.target;
		var search = {};

		if (this.state.type !== '') {
			search.type = this.state.type;
		}

		if (id === 'search') {
			this.setState({ searchWord: value });
			if (Number.isInteger(Number(value))) {
				search.pokedex_number = value;
			} else {
				search.name = value;
			}
		} else if (id === 'order') {
			if (value !== '') {
				search.filter = value.split('-')[0];
				search.order = value.split('-')[1];
			}
			search.name = this.state.searchWord;
		}
		api.get('/', {
			params: search,
		}).then(response => {
			this.setState({
				data: response.data.rows,
				rows: response.data.count,
			});
		});
		window.history.pushState({}, '/', '/');
	}

	typeFilter(btn) {
		const { target } = btn;
		const { types } = target.attributes;

		var params = {};
		params.type = types.value;
		if (this.state.searchWord) {
			if (Number.isInteger(Number(this.state.searchWord))) {
				params.pokedex_number = this.state.searchWord;
			} else {
				params.name = this.state.searchWord;
			}
		}

		api.get('/', {
			params,
		}).then(response => {
			this.setState({ type: types.value });
			this.setState({
				data: response.data.rows,
				rows: response.data.count,
			});
		});
	}

	clear(event) {
		this.setState({ type: '', searchWord: '' });
		api.get('/').then(response => {
			this.setState({
				data: response.data.rows,
				rows: response.data.count,
			});
		});
	}
	clearWord() {
		this.setState({ searchWord: '' });
		api.get('/', {
			params: { type: this.state.type },
		}).then(response => {
			this.setState({
				data: response.data.rows,
				rows: response.data.count,
			});
		});
	}
	clearType() {
		this.setState({ type: '' });

		var params = {};

		if (this.state.searchWord) {
			if (Number.isInteger(Number(this.state.searchWord))) {
				params.pokedex_number = this.state.searchWord;
			} else {
				params.name = this.state.searchWord;
			}
		}
		api.get('/', { params }).then(response => {
			let result = response.data.rows;
			this.setState({ data: result });
		});
	}

	render() {
		return (
			<Main>
				<Search>
					<Container>
						<Row>
							<Col
								xs={12}
								className='d-flex justify-content-center'>
								<Col xs={12} lg={6}>
									<InputGroup>
										<FormControl
											onChange={event =>
												this.search(event)
											}
											type='text'
											id='search'
											placeholder='Pokemon ou ID'
											value={this.state.searchWord}
										/>
									</InputGroup>
									<Row>
										{this.state.searchWord ? (
											<Col xs={3} className='mt-2'>
												<Button
													onClick={event =>
														this.clearWord()
													}>
													{this.state.searchWord}
												</Button>
											</Col>
										) : (
											<> </>
										)}
										{this.state.type ? (
											<Col xs={3} className='mt-2'>
												<BtnFilter
													types={`${this.state.type}`}
													onClick={event =>
														this.clearType(event)
													}>
													{this.state.type}
												</BtnFilter>
											</Col>
										) : (
											<> </>
										)}
										<Button
											variant='link'
											onClick={event =>
												this.clear(event)
											}>
											{' '}
											Limpar{' '}
										</Button>
									</Row>
								</Col>
							</Col>
						</Row>
					</Container>
				</Search>
				<Filter>
					<Container>
						<Col xs={12}>
							<Accordion className='d-flex justify-content-center text-center'>
								<Card>
									<Card.Header>
										<Accordion.Toggle
											as={Button}
											variant='dark'
											eventKey='0'>
											Busca Avan√ßada
										</Accordion.Toggle>
									</Card.Header>
									<Accordion.Collapse eventKey='0'>
										<Card.Body>
											<Row className='justify-content-center types_filter'>
												<Col xs={12} sm={4} md={3}>
													<Col>
														<BtnFilter
															types='grass'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Grass
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='fire'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Fire
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='water'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Water
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='bug'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Bug
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='normal'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Normal
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='poison'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Poison
														</BtnFilter>
													</Col>
												</Col>
												<Col xs={12} sm={4} md={3}>
													<Col>
														<BtnFilter
															types='electric'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															Electric
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='ground'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															ground
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='fairy'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															fairy
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='fighting'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															fighting
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='psychic'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															psychic
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='rock'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															rock
														</BtnFilter>
													</Col>
												</Col>
												<Col xs={12} sm={4} md={3}>
													<Col>
														<BtnFilter
															types='ghost'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															ghost
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='ice'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															ice
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='dragon'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															dragon
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='dark'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															dark
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='steel'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															steel
														</BtnFilter>
													</Col>
													<Col>
														<BtnFilter
															types='flying'
															onClick={event =>
																this.typeFilter(
																	event
																)
															}>
															flying
														</BtnFilter>
													</Col>
												</Col>
											</Row>
											<Row></Row>
										</Card.Body>
									</Accordion.Collapse>
								</Card>
							</Accordion>
						</Col>
					</Container>
				</Filter>
				<Container className='mt-2'>
					<Row className='justify-content-end'>
						<Col xs={2}>
							<FormControl
								as='select'
								id='order'
								onChange={event => this.search(event)}>
								<option value=''> Ordenar por </option>
								<option value='pokedex_number-ASC'>
									Menor Numero
								</option>
								<option value='pokedex_number-DESC'>
									Maior Numero
								</option>
								<option value='name-ASC'> A-Z </option>
								<option value='name-DESC'> Z-A </option>
							</FormControl>
						</Col>
					</Row>
				</Container>
				<Container className='mt-5'>
					<Row>
						{this.state.data.map(pokemon => (
							<Col
								xs={12}
								sm={6}
								md={4}
								lg={3}
								xl={3}
								key={pokemon.id}
								className='mb-3'>
								<Flip top>
									<Pokemons type={pokemon.type1.name}>
										<Link to={'/pokemon/' + pokemon.name}>
											<h3>
												{pad(pokemon.pokedex_number, 3)}
											</h3>

											<Image
												src={
													'https://assets.pokemon.com/assets/cms2/img/pokedex/detail/' +
													pad(
														pokemon.pokedex_number,
														3
													) +
													'.png'
												}
											/>
											<h2> {pokemon.name} </h2>
										</Link>
										<Container>
											<Types type={pokemon.type1.name}>
												<Type
													type={pokemon.type1.name}
													first='true'>
													<Link
														to={
															'/type/' +
															pokemon.type1.name
														}>
														{pokemon.type1.name}
													</Link>
												</Type>

												{pokemon.type2 !== null ? (
													<Type
														type={
															pokemon.type2.name
														}>
														<Link
															to={
																'/type/' +
																pokemon.type2
																	.name
															}>
															{pokemon.type2.name}
														</Link>
													</Type>
												) : (
													<> </>
												)}
											</Types>
										</Container>
									</Pokemons>
								</Flip>
							</Col>
						))}
					</Row>
				</Container>
				{this.state.rows > 20 ? (
					<Pagination rows={this.state.rows} current={0} />
				) : (
					<> </>
				)}
			</Main>
		);
	}
}
