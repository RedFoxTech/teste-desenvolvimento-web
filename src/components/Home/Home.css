.homeButton img {
    width: 40px;
    position: absolute;
    top: 15px;
    left: 15px;
}

@media (max-width: 500px) {
    .homeButton img {
        width: 30px;
    }
}