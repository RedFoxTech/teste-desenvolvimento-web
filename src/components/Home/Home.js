import React from 'react';

import './Home.css';

import Ash from '../../assets/home.png';

const Home = (props) => {
    return(
        <a href="/" className="homeButton"><img src={Ash} alt="home button"/></a>
    );
}

export default Home;