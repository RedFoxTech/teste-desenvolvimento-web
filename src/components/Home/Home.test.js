import React from 'react';
import Home from './Home';

let wrapper;
beforeEach(() => {
    wrapper = shallow(<Home/>);
})

describe("rendering test/snapshot", () => {
    it('should render a pokeball button', () => {
        expect(wrapper).toMatchSnapshot()
    })
})