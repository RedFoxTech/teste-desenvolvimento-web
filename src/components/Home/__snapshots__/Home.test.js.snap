// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test/snapshot should render a pokeball button 1`] = `
<a
  className="homeButton"
  href="/"
>
  <img
    alt="home button"
    src="home.png"
  />
</a>
`;
